EmpireLegends/GameStart.cpp

```diff
@@ -0,0 +1,198 @@
+
+		std::cout << "\nEnter player name 2: ";
+		getline(std::cin, playerName2);
+		p2 = new Player(playerName2, playerCoins);
+		std::cout << "Welcome " << playerName2 << " !" << std::endl;
+
+		std::cout << "\nEach player gets 14 coins." << std::endl;
+
+		break;
+
+	case 3:
+
+		playerCoins = 11;
+
+		std::cout << "Enter player name 1: ";
+		std::cin.ignore();
+		getline(std::cin, playerName1);
+		p1 = new Player(playerName1, playerCoins);
+		std::cout << "Welcome " << playerName1 << " !" << std::endl;
+
+		std::cout << "\nEnter player name 2: ";
+		getline(std::cin, playerName2);
+		p2 = new Player(playerName2, playerCoins);
+		std::cout << "Welcome " << playerName2 << " !" << std::endl;
+
+		std::cout << "\nEnter player name 3: ";
+		getline(std::cin, playerName3);
+		p3 = new Player(playerName3, playerCoins);
+		std::cout << "Welcome " << playerName3 << " !" << std::endl;
+
+		std::cout << "\nEach player gets 11 coins." << std::endl;
+
+		break;
+
+	case 4:
+
+		playerCoins = 9;
+
+		std::cout << "Enter player name 1: ";
+		std::cin.ignore();
+		getline(std::cin, playerName1);
+		p1 = new Player(playerName1, playerCoins);
+		std::cout << "Welcome " << playerName1 << " !" << std::endl;
+
+		std::cout << "\nEnter player name 2: ";
+		getline(std::cin, playerName2);
+		p2 = new Player(playerName2, playerCoins);
+		std::cout << "Welcome " << playerName2 << " !" << std::endl;
+
+		std::cout << "\nEnter player name 3: ";
+		getline(std::cin, playerName3);
+		p3 = new Player(playerName3, playerCoins);
+		std::cout << "Welcome " << playerName3 << " !" << std::endl;
+
+		std::cout << "\nEnter player name 4: ";
+		getline(std::cin, playerName4);
+		p4 = new Player(playerName4, playerCoins);
+		std::cout << "Welcome " << playerName4 << " !" << std::endl;
+
+		std::cout << "\nEach player gets 9 coins." << std::endl;
+
+		break;
+
+	default:
+		while (numOfPlayers > 4 || numOfPlayers < 2) {
+			std::cout << "INVALID number of players." << std::endl;
+			std::cout << "Please select a number between 2 and 4: ";
+			std::cin >> numOfPlayers;
+		}
+	}
+}
+
+void GameStart::playerBidding() {
+	biddingFacility = new BiddingFacility();
+
+	std::cout << "Entering the bidding phase with " << numOfPlayers << " players..." << std::endl;
+
+	// Allow players to place their bids
+	switch (numOfPlayers) {
+	case 2:
+
+		biddingPlayers.clear();
+		biddingPlayers.push_back(p1);
+		biddingPlayers.push_back(p2);
+
+		biddingFacility->placeBids(biddingPlayers);
+
+		break;
+
+	case 3:
+
+		biddingPlayers.clear();
+		biddingPlayers.push_back(p1);
+		biddingPlayers.push_back(p2);
+		biddingPlayers.push_back(p3);
+
+		biddingFacility->placeBids(biddingPlayers);
+
+		break;
+
+	case 4:
+
+		biddingPlayers.clear();
+		biddingPlayers.push_back(p1);
+		biddingPlayers.push_back(p2);
+		biddingPlayers.push_back(p3);
+		biddingPlayers.push_back(p4);
+
+		biddingFacility->placeBids(biddingPlayers);
+
+		break;
+
+	default:
+		biddingPlayers.clear();
+	}
```

Comment:
since the bidding logic isn't dependent on the number of players, I feel like this switch can just be replaced by a loop?

Like instead of having p1...p4 we can just have a vector

Reply:
sounds like a great idea! i agree that having a single vector would simplify the whole player name/ player pointer business 👍 and then instead of this loop, we can just put it in `detPlayerCount` and that'll also simplify `playerBidding` a lot